runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/SdkHarnessClient.java

```diff
@@ -273,58 +324,120 @@ public SdkHarnessClient withIdGenerator(IdGenerator idGenerator) {
+   */
   public <T> BundleProcessor<T> getProcessor(
       BeamFnApi.ProcessBundleDescriptor descriptor,
       RemoteInputDestination<WindowedValue<T>> remoteInputDesination) {
-    try {
-      return clientProcessors.get(
-          descriptor.getId(),
-          () ->
-              (BundleProcessor)
-                  register(
-                          Collections.singletonMap(
-                              descriptor, (RemoteInputDestination) remoteInputDesination))
-                      .get(descriptor.getId()));
-    } catch (ExecutionException e) {
-      throw new RuntimeException(e);
+    checkState(!descriptor.hasStateApiServiceDescriptor(),
+        "The %s cannot support a %s containing a state %s.",
+        BundleProcessor.class.getSimpleName(),
+        BeamFnApi.ProcessBundleDescriptor.class.getSimpleName(),
+        Endpoints.ApiServiceDescriptor.class.getSimpleName());
+    return getProcessor(descriptor, remoteInputDesination, NoOpStateDelegator.INSTANCE);
+  }
+
+  /**
+   * A {@link StateDelegator} that issues zero state requests to any provided
+   * {@link StateRequestHandler state handlers}.
+   */
+  private static class NoOpStateDelegator implements StateDelegator {
+    private static final NoOpStateDelegator INSTANCE = new NoOpStateDelegator();
+    @Override
+    public Registration registerForProcessBundleInstructionId(String processBundleInstructionId,
+        StateRequestHandler handler) {
+      return Registration.INSTANCE;
+    }
+
+    /**
+     * The corresponding registration for a {@link NoOpStateDelegator} that does nothing.
+     */
+    private static class Registration implements StateDelegator.Registration {
+      private static final Registration INSTANCE = new Registration();
+
+      @Override
+      public void deregister() {
+      }
+
+      @Override
+      public void abort() {
+      }
     }
   }
 
   /**
-   * Registers a {@link BeamFnApi.ProcessBundleDescriptor} for future processing.
+   * Provides {@link BundleProcessor} that is capable of processing bundles containing
+   * state accesses such as:
+   * <ul>
+   *   <li>Side inputs</li>
+   *   <li>User state</li>
+   *   <li>Remote references</li>
+   * </ul>
    *
-   * <p>A client may block on the result future, but may also proceed without blocking.
+   * <p>Note that bundle processors are cached based upon the the
+   * {@link ProcessBundleDescriptor#getId() process bundle descriptor id}.
+   * A previously created instance may be returned.
    */
-  public Map<String, BundleProcessor> register(
-      Map<BeamFnApi.ProcessBundleDescriptor, RemoteInputDestination<WindowedValue<?>>>
-          processBundleDescriptors) {
+  public <T> BundleProcessor<T> getProcessor(
+      BeamFnApi.ProcessBundleDescriptor descriptor,
+      RemoteInputDestination<WindowedValue<T>> remoteInputDesination,
+      StateDelegator stateDelegator) {
+    BundleProcessor<T> bundleProcessor = clientProcessors.computeIfAbsent(
```

Comment:
prefer breaking the line at the higher syntactic level.  In this case after the '='

Reply:
Done and one other place.